docs_source/usage/vocabs-unique-keys.md

```diff
@@ -0,0 +1,128 @@
+# A Vocabulary for Identifying Uniqueness of Array Items
+
+## 1. Purpose
+
+This document describes a vocabulary defining keywords that can be used to validate that arrary instances contain unique values as determined by one or more given paths within each item.
+
+The intent for these keywords is to cover the use cases discussed in the JSON Schema vocabularies GitHub issue [#22](https://github.com/json-schema-org/json-schema-vocabularies/issues/22).
+
+## 2. Declarations
+
+The ID for this vocabulary is `https://gregsdennis.github.io/json-everything/vocabs-unique-keys` (the URI to this document).
+
+A draft 2020-12 meta-schema which includes this vocabulary has been defined for convenience.  The ID for the meta-schema is `https://gregsdennis.github.io/json-everything/meta/unique-keys`, and it can also be found at this address.
+
+## 3. The `uniqueKeys` Keyword
+
+### 3.1 Syntax and Semantics
+
+The value of `uniqueKeys` must be an array of JSON Pointers as defined by [RFC 6901](https://tools.ietf.org/html/rfc6901).  At least one JSON Pointer MUST be specified.
+
+These JSON Pointers, when resolved against each item, define a set of values for that item.  If an item does not contain the location specified by a JSON Pointer, the implementation MUST keep a placeholder to indicate a missing value.
+
+Validation for this keyword passes if the value sets for all of the items are distinct, that is, no two sets contain the same collection, including missing values.
```

Comment:
It might be useful to produce an annotation that consists of these value sets -- it could be useful for debugging why something unexpectedly passed validation.

Reply:
Do we have other keywords that validate _and_ annotate?

Reply:
Most of the applicator keywords?  I don't think it's relevant that they use subschemas as part of their validation -- they still produce a true/false result based on their own unique rules.

Reply:
I agree with this in principal, but...

1. I'm struggling to find a way for the implementation to represent a missing value.
2. "Unexpectedly passes" would be the only benefit; "unexpectedly fails" wouldn't work because a failure (which I think is the more interesting case) would omit annotations.
3. The annotation would be an array of values _for each item_.  This could get very large very quickly.